.github/workflows/test.yml

```diff
@@ -1,26 +1,41 @@
-on: push
+on:
+  push:
+    branches:
+      - main
+      - staging
+      - trying
+      - release/**
+  pull_request:
+  merge_group:
+
 name: Run tests
 jobs:
-  # The `ci-result` job doesn't actually test anything - it just aggregates the
-  # overall build status for bors, otherwise our bors.toml would need an entry
+  # The `ci-success` job doesn't actually test anything - it just aggregates the
+  # overall build status, otherwise the merge queue would need an entry
   # for each individual job produced by the job-matrix.
   #
-  # Ref: https://github.com/rust-lang/crater/blob/9ab6f9697c901c4a44025cf0a39b73ad5b37d198/.github/workflows/bors.yml#L125-L149
-  #
-  # ALL THE SUBSEQUENT JOBS NEED THEIR `name` ADDED TO THE `needs` SECTION OF THIS JOB!
-  ci-result:
+  # ALL THE SUBSEQUENT JOBS NEED THEIR `name` ADDED TO THE `needs` SECTION OF both "ci result" JOBS!
+  ci-success:
     name: ci result
     runs-on: ubuntu-latest
     needs:
       - rstar
       - check
       - no_std
+    if: success()
     steps:
       - name: Mark the job as a success
-        if: success()
         run: exit 0
+  ci-failure:
```

Comment:
It is somewhat reasonable, but just to have the alternative discussed here, we could also have a single job dynamically checking the other jobs results, e.g. in PyO3 we use

```yaml
conclusion:
needs:
    - fmt
    - check-msrv
    - clippy
    - build-pr
    - build-full
    - valgrind
    - careful
    - coverage
    - emscripten
if: always()
runs-on: ubuntu-latest
steps:
    - name: Result
    run: |
        jq -C <<< "${needs}"
        # Check if all needs were successful or skipped.
        "$(jq -r 'all(.result as $result | (["success", "skipped"] | contains([$result])))' <<< "${needs}")"
    env:
        needs: ${{ toJson(needs) }}
```

Reply:
Alternatively, we could at least factor out the job list using a YAML anchor?

Reply:
Certainly more DRY.